yarn-project/end-to-end/src/e2e_fees/private_refunds.test.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  type AztecAddress,
+  ExtendedNote,
+  type FeePaymentMethod,
+  type FunctionCall,
+  Note,
+  type Wallet,
+} from '@aztec/aztec.js';
+import { Fr, type GasSettings } from '@aztec/circuits.js';
+import { FunctionSelector, FunctionType } from '@aztec/foundation/abi';
+import { type PrivateFPCContract, PrivateTokenContract } from '@aztec/noir-contracts.js';
+
+import { expectMapping } from '../fixtures/utils.js';
+import { FeesTest } from './fees_test.js';
+
+describe('e2e_fees/private_refunds', () => {
+  let aliceWallet: Wallet;
+  let aliceAddress: AztecAddress;
+  let privateToken: PrivateTokenContract;
+  let privateFPC: PrivateFPCContract;
+
+  let InitialAlicePrivateTokens: bigint;
+  let InitialBobPrivateTokens: bigint;
+  let InitialPrivateFPCGas: bigint;
+
+  const t = new FeesTest('private_refunds');
+
+  beforeAll(async () => {
+    await t.applyInitialAccountsSnapshot();
+    await t.applyPublicDeployAccountsSnapshot();
+    await t.applyDeployGasTokenSnapshot();
+    await t.applyPrivateTokenAndFPC();
+    await t.applyFundAliceWithPrivateTokens();
+    ({ aliceWallet, aliceAddress, privateFPC, privateToken } = await t.setup());
+    t.logger.debug(`Alice address: ${aliceAddress}`);
+  });
+
+  afterAll(async () => {
+    await t.teardown();
+  });
+
+  beforeEach(async () => {
+    [[InitialAlicePrivateTokens, InitialBobPrivateTokens], [InitialPrivateFPCGas]] = await Promise.all([
+      t.privateTokenBalances(aliceAddress, t.bobAddress),
+      t.gasBalances(privateFPC.address),
+    ]);
+  });
+
+  it('can do private payments and refunds', async () => {
+    const bobKeyHash = t.bobWallet.getCompleteAddress().publicKeys.masterNullifierPublicKey.hash();
+    const rebateNonce = new Fr(42);
+    const tx = await privateToken.methods
```

Comment:
Do we submit this tx just to get the tx fee? And if yes is it because we have a guarantee of the fee being constant between txs?

Reply:
I believe it is submitted so we can test the refund flow. If we didn't submit it, the public execution that creates the refund wouldn't run.

Reply:
Oh, got it. Will add a comment there. Thank you 🙏